autoPyTorch/pipeline/components/setup/network_backbone/ShapedResNetBackbone.py

```diff
@@ -40,8 +40,12 @@ def build_backbone(self, input_shape: Tuple[int, ...]) -> None:
         )
         if self.config['use_dropout'] and self.config["max_dropout"] > 0.05:
             dropout_shape = get_shaped_neuron_counts(
-                self.config['dropout_shape'], 0, 0, 1000, self.config['num_groups']
-            )
+                self.config['dropout_shape'],
+                0,
+                self.config["max_dropout"],
+                1000,
+                self.config['num_groups'] + 2,
+            )[:-1]
```

Comment:
why are we leaving out the last of the shape? Wouldn't it mean that it won't have a dropout value for the last block?

Reply:
Hmm no, as the number of layers which is passed is `self.config['num_groups'] + 2`. Thinking about it more, this should actually be +1, since dropout is not applied after the first input layer, but only between the layers between the `ResBlock`.
The reason why we pass one more and we skip the last one, is because here https://github.com/automl/Auto-PyTorch/blob/2b633cec1d44f8ca7f1126f2c9ca2ca987cfdba8/autoPyTorch/pipeline/components/setup/network_backbone/utils.py#L133 at all shapes in the end we pass `output_features`. We actually want a shape and output features in the probably author implementation is meant for the number of classes. So, we can pass 0 for the output features to actually have the shapes and never actually include it, because it would not satisfy the shapes, for example if we want a `triangle` shape, including the last count, would mean including 0, which breaks the implementation.

Reply:
okay that makes sense then

Reply:
btw could you add this as a comment above the code.

Reply:
Good idea, I also had to look at it to understand what was happening.

Reply:
I am also going to remove the max_dropout from the if condition above, since I think it serves no purpose and it will give a crash if it does not go through it because of that.

Reply:
yeas that makes sense